www/gatsby-node.js

```diff
@@ -580,6 +585,24 @@ exports.onCreateNode = ({ node, actions, getNode, reporter }) => {
   // end Community/Creators Pages
 }
 
+exports.onCreatePage = ({ page, actions }) => {
+  // add lists of featured items to Ecosystem page
+  if (page.path === "/ecosystem/") {
+    const { createPage, deletePage } = actions
+    return new Promise(resolve => {
```

Comment:
Not sure a Promise is needed here!

Reply:
I don't believe this section will be necessary -> https://github.com/gatsbyjs/gatsby/pull/9442/files/c74e8d61c236371c553b1d634ecbca8d371544cb#diff-fb043578f52339bcb2a2d2b4df1203c8

Reply:
@DSchau 
When I remove the Promise like that
```
exports.onCreatePage = ({ page, actions }) => {
  // add lists of featured items to Ecosystem page
  if (page.path === "/ecosystem/") {
    const { createPage, deletePage } = actions
    const oldPage = Object.assign({}, page)

    page.context.featuredStarters = ecosystemFeaturedItems.starters
    page.context.featuredPlugins = ecosystemFeaturedItems.plugins

    deletePage(oldPage)
    createPage(page)
  }
}
```
it works, but could you give me a hint why there is a promise in the docs
https://www.gatsbyjs.org/docs/creating-and-modifying-pages/#modifying-pages-created-by-core-or-plugins? For me it seems like similar case.

@amberleyromo

Reply:
@greglobinski that's a great point. To me, that example should be re-written, there's nothing async required there as far as I can tell.

Reply:
:+1: